build/fluent_loader.js

```diff
@@ -20,34 +12,45 @@ module.exports = function(source) {
   if (!locale) {
     throw new Error(`couldn't find locale in: ${this.resourcePath}`);
   }
-  // load default language and "merge" contexts
-  // TODO: make this configurable
-  const en_ftl = fs.readFileSync(
-    require.resolve('../public/locales/en-US/send.ftl'),
-    'utf8'
-  );
-  const en = new FluentBundle('en-US');
-  en.addMessages(en_ftl);
-  // pre-parse the ftl
-  const context = new FluentBundle(locale);
-  context.addMessages(source);
 
-  const merged = merge(en._messages, context._messages);
+  // Parse the current language's translation file.
+  const locResource = FluentResource.fromString(source);
+  let enResource;
+
+  // If the current language is not en-US, also parse en-US to provide a
+  // fallback for missing translations.
+  if (locale !== 'en-US') {
+    const en_ftl = fs.readFileSync(
+      require.resolve('../public/locales/en-US/send.ftl'),
+      'utf8'
+    );
+    enResource = FluentResource.fromString(en_ftl);
+  }
+
   return `
 module.exports = \`
 if (typeof window === 'undefined') {
   var fluent = require('fluent');
 }
 (function () {
-  var bundle = new fluent.FluentBundle('${locale}', {useIsolating: false});
-  bundle._messages = new Map(${toJSON(merged)});
+  let bundles = [
+    ['${locale}', ${toJSON(locResource)}],
+    ${enResource ? `['en-US', ${toJSON(enResource)}]` : ''}
+  ].map(([locale, entries]) => {
+    let bundle = new fluent.FluentBundle(locale, {useIsolating: false});
+    bundle.addResource(new fluent.FluentResource(entries));
+    return bundle;
+  });
+
   function translate(id, data) {
-    var msg = bundle.getMessage(id);
-    if (typeof(msg) !== 'string' && !msg.val && msg.attrs) {
-      msg = msg.attrs.title || msg.attrs.alt
+    for (let bundle of bundles) {
+      if (bundle.hasMessage(id)) {
+        let message = bundle.getMessage(id);
+        return bundle.format(message, data);
```

Comment:
I didn't see any messages in the current FTL file which use `title` or `alt`. We're going to change the `format` API soon to allow formatting attributes, too (probably via `format("string-id.attribute-name")`). I only implemented formatting message values here. If formatting attributes is required now, I can add it in this PR. Otherwise, I suggest to wait for the next release of `fluent`.

Reply:
we don't use any attributes atm so 👍